tests/page/Switch.test.js

```diff
@@ -0,0 +1,113 @@
+import { URL_MATCH_FAILURE_TYPE } from '../../src/actions.js'
+
+configure({ adapter: new Adapter() });
+
+const baseProps = {
+  options: {},
+  url: '',
+  dispatch: () => {},
+};
+
+const buildComponentInstance = (props = {}) => shallow(<Switch { ...baseProps } { ...props } />).instance();
+
+describe('page/Switch', () => {
+  it('is React component', () => {
+    expect(Switch.prototype).to.be.an.instanceof(Component);
+  });
+});
+
+describe('page/Switch.resolve', () => {
+  it('is a function', () => {
+    const instance = buildComponentInstance();
+    expect(instance.resolve).to.be.an('function');
+  });
+
+  it('returns React component', () => {
+    const instance = buildComponentInstance();
+    expect(React.isValidElement(instance.resolve())).to.equal(true);
+  });
+
+  it('calls route resolver', () => {
+    const resolver = spy(() => {});
+
+    const instance = buildComponentInstance({
+      routeResolver: resolver,
+    });
+
+    instance.resolve();
+    expect(resolver.called).to.equal(true);
+  });
+
+  it('calls route resolver with options passed through', () => {
+    const resolver = spy(() => {});
+    const options = {
+      someValue: { url: '', action: '' },
+    };
+
+    const instance = buildComponentInstance({
+      routeResolver: resolver,
+      options,
+    });
+
+    instance.resolve();
+    expect(resolver.getCall(0).args[0]).to.deep.equal(options);
+  });
+
+  it('calls url matcher', () => {
+    const matcher = spy(() => {});
+    const instance = buildComponentInstance({
+      matcher,
+    });
+
+    instance.resolve();
+    expect(matcher.called).to.equal(true);
+  });
+
+  it('calls dispatch with URL_MATCH_FAILURE_TYPE action when option had not been found', () => {
+    const dispatch = spy(() => {});
+    const instance = buildComponentInstance({
+      options: {},
+      url: '/some/not/matching/url',
+      dispatch,
+    });
+
+    instance.resolve();
+    expect(dispatch.called).to.equal(true);
+    expect(dispatch.getCall(0).args[0]).to.have.property('type', URL_MATCH_FAILURE_TYPE);
+  });
+
+  it('does not call dispatch when option had been found', () => {
+    const dispatch = spy(() => {});
+    const instance = buildComponentInstance({
+      options: {
+        'route': { url: '/some/matching/url', action: 'list', }
+      },
+      url: '/some/matching/url/list',
+      dispatch,
+    });
+
+    instance.resolve();
+    expect(dispatch.called).to.equal(false);
+  });
+
+  it('calls dispatch with url, options and error in payload when option had not been found', () => {
+    const dispatch = spy(() => {});
+    const instance = buildComponentInstance({
+      options: {},
+      url: '/some/not/matching/url',
+      dispatch,
+    });
+
+    instance.resolve();
+    expect(dispatch.called).to.equal(true);
+    expect(dispatch.getCall(0).args[0]).to.have.property('payload');
+    expect(dispatch.getCall(0).args[0].payload).to.have.keys('url', 'options', 'error');
+  });
+});
```

Comment:
I don't fully understand those test, please spend a few minutes with me, to make them clear.